packages/documentsearch/src/searchinstance.ts

```diff
@@ -12,8 +12,8 @@ import { Widget } from '@phosphor/widgets';
 /**
  * Represents a search on a single widget.
  */
-export class SearchInstance implements IDisposable {
-  constructor(widget: Widget, searchProvider: ISearchProvider) {
+export class SearchInstance<T extends Widget = Widget> implements IDisposable {
+  constructor(widget: T, searchProvider: ISearchProvider<T>) {
```

Comment:
Do we actually benefit from making this generic, or could we just use the "base" version of `ISearchProvider` here?

Reply:
Apparently not, since it'll still compile without it being generic. Originally `ISearchProvider` didn't have a default template arg, so I had thought that `SearchInstance` had to be generic too.

I've made `SearchInstance` non-generic.